homeassistant/components/somfy/climate.py

```diff
@@ -0,0 +1,214 @@
+        return self._climate.get_humidity()
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if self._climate.get_regulation_state() == RegulationState.TIMETABLE:
+            return HVAC_MODE_AUTO
+        return HVAC_MODES_MAPPING.get(self._climate.get_hvac_state())
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes.
+
+        HEAT and COOL mode are exclusive. End user has to enable a mode manually within the Somfy application.
+        So only one mode can be displayed. Auto mode is a scheduler.
+        """
+        hvac_state = HVAC_MODES_MAPPING.get(self._climate.get_hvac_state())
+        return [HVAC_MODE_AUTO, hvac_state]
+
+    def set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        if hvac_mode == self.hvac_mode:
+            return
+        if hvac_mode == HVAC_MODE_AUTO:
+            self._climate.cancel_target()
+        else:
+            self._climate.set_target(
+                TargetMode.MANUAL, self.target_temperature, DurationType.FURTHER_NOTICE
+            )
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation if supported."""
+        if not self.current_temperature or not self.target_temperature:
+            return CURRENT_HVAC_IDLE
+
+        if (
+            self.hvac_mode == HVAC_MODE_HEAT
+            and self.current_temperature < self.target_temperature
+        ):
+            return CURRENT_HVAC_HEAT
+
+        if (
+            self.hvac_mode == HVAC_MODE_COOL
+            and self.current_temperature > self.target_temperature
+        ):
+            return CURRENT_HVAC_COOL
+
+        return CURRENT_HVAC_IDLE
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode."""
+        mode = self._climate.get_target_mode()
+        return PRESETS_MAPPING.get(mode)
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return a list of available preset modes."""
+        return list(PRESETS_MAPPING.values())
+
+    def set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        if self.preset_mode == preset_mode:
+            return
+
+        if preset_mode == PRESET_HOME:
+            temperature = self._climate.get_at_home_temperature()
+        elif preset_mode == PRESET_AWAY:
+            temperature = self._climate.get_away_temperature()
+        elif preset_mode == PRESET_SLEEP:
+            temperature = self._climate.get_night_temperature()
+        elif preset_mode == PRESET_FROST_GUARD:
+            temperature = self._climate.get_frost_protection_temperature()
+        elif preset_mode in [PRESET_MANUAL, PRESET_GEOFENCING]:
+            temperature = self.target_temperature
+        else:
+            _LOGGER.error("Preset mode not supported: %s", preset_mode)
```

Comment:
We normally raise a `ValueError` with this message as exception argument.

Reply:
OK